src/mvba/broadcaster.rs

```diff
@@ -2,55 +2,60 @@ use super::{bundle::Bundle, NodeId};
 use blsttc::{PublicKeyShare, SecretKeyShare};
 
 // Broadcaster holds information required to broadcast the messages.
-// If a node is an observer node, it doesn't broadcast messages.
-// For observer node SecretKeyShare and Node ID set to None
+// If a node is an observer node, it should't broadcast messages.
+// TODO: How to find out it is an observer node or not?
```

Comment:
For simplicity of code, I'd drop the observer role entirely. Fewer branches to think about and we could always add it back later if we find a need

Reply:
> Coming along +1, it looks like the implementation is nearly there, it seems like we're handling all messages up to the final signature aggregation and `c-final` broadcast.

Yes, we are in the right track

> Comments are mostly nits

Will do